docs/investigation/composed-action/REAMDE.md

```diff
@@ -0,0 +1,163 @@
+    ```
+
+13. Extract the output TypeInstances:
+    ```
+    capact action get psql-install -ojson | jq -r '.Actions[].output.typeInstances | map(select(.typeRef.path == "cap.type.database.postgresql.config"))'
+    ```
+
+
+And if you want to use the created PostgreSQL database for other Action, you need to repeat all those steps manually. For now, we overcome this issue by creating Implementation "umbrella workflows". This, unfortunately, results in an additional boilerplate and cause that steps are not atomic causing problem such as https://github.com/capactio/capact/issues/695 or https://github.com/capactio/capact/issues/563. Additionally, such approach make the individual Action less reusable as they don't have a single responsibility.
+
+### ComposedAction
+
+The main responsibility of the Composed Action CR is to schedule a set of Action, watch their execution and pass the input parameters between the steps.
+
+```yaml
+apiVersion: core.capact.io/v1alpha1
+kind: ComposedAction
+metadata:
+  name: "full-spec"
+spec:
+  steps:
+    psql:
+      interface:
+        path: "cap.interface.database.postgresql.install"
+      input:
+        input-parameters:
+          raw:
+            data: |
+              superuser:
+                username: superuser
+              defaultDBName: postgres
+    create-user:
+      depends: ["psql"] # this can be removed, we can create own DAG based on the inputs.
+      interface:
+        path: "cap.interface.database.postgresql.create-user"
+      input:
+       postgresql:
+         from: psql.postgresql
+       input-parameters:
+         raw:
+           data: |
+             name: mattermost
+
+    create-db:
+      depends: ["create-user"]
+      interface:
+        path: "cap.interface.database.postgresql.create-user"
+      input:
+        postgresql:
+          from: psql.postgresql
+        input-parameters:
+          raw:
+            data: |
+              name: mattermost
+              owner: mattermost
+    install:
+      depends: ["create-db"]
+      interface:
+        path: "cap.interface.productivity.mattermost.install"
+      input:
+        postgresql:
+          from: psql.postgresql
+        database:
+          from: create-db.database
+        user:
+          from: create-user.user
+  advancedRendering:
+    enabled: false
+  run: false
+  dryRun: false
+  cancel: false
+```
+
+In the future, we can also add a native support for direct Implementation references in steps. When user compose an Action, they already know what they want to achieve. Instead of creating a complex Action Policy, specific Implementations can be selected with an implementation-specific input attached to them. In this way, the Interfaces would be a nice grouping feature used on UI and later user by browsing a catalog of available service, can compose a given Action. Such Action will be easier to rationalize and requires less cognitive overhead as it describes in a declarative way want will be done.
```

Comment:
```suggestion
In the future, we can also add a native support for direct Implementation references in steps. When users compose an Action, they already know what they want to achieve. Instead of creating a complex Action Policy, specific Implementations can be selected with an implementation-specific input attached to them. In this way, the Interfaces would be a nice grouping feature used on UI and later user by browsing a catalog of available service, can compose a given Action. Such Action will be easier to rationalize and requires less cognitive overhead as it describes in a declarative way want will be done.
```